table/block_based/index_builder.h

```diff
@@ -658,8 +673,16 @@ class PartitionedIndexBuilder : public IndexBuilder {
   size_t TopLevelIndexSize(uint64_t) const { return top_level_index_size_; }
   size_t NumPartitions() const;
 
-  // TODO: implement
-  uint64_t EstimateCurrentIndexSize() const override { return 0; }
+  // Returns a cached estimate of the current index size. This
+  // estimate is updated when data blocks are added.
+  uint64_t CurrentIndexSizeEstimate() const override {
+    uint64_t estimate = estimated_index_size_.LoadRelaxed();
+    printf(
```

Comment:
Noting debug output. Btw, I recommend `fprintf(stderr, ...)` because it is more consistently printed under unit tests